Modules/md5module.c

```diff
@@ -349,9 +359,10 @@ static PyObject *
 MD5Type_copy_impl(MD5object *self)
 /*[clinic end generated code: output=596eb36852f02071 input=2c09e6d2493f3079]*/
 {
-    MD5object *newobj;
+    MD5State *st = PyType_GetModuleState(Py_TYPE(self));
```

Comment:
You should use "cls: defining_class" in the clinic input, and use cls rather than Py_TYPE(self).

So you can remove the comment below:

```
// Using PyType_GetModuleState() on this type is safe since
// it cannot be subclassed: it does not have the Py_TPFLAGS_BASETYPE flag.
```

I guess that the two other modules can get a similar change.

Reply:
@vstinner sure but why would I remove the comment.  I would just replace it with PyType_GetModuleState(cls), right?  Isn't the comment still relevant?

Reply:
If you never use `Py_TYPE(self)` but only pass the defining class, subclassing the type is ok and so the comment can be removed.

Reply:
Please remove the comment.

Reply:
@vstinner oh!  I didn't know that.  This means that I could probably continue with some of the other modules with Py_TPFLAGS_BASETYPE!

Reply:
Sadly, argument clinic cannot be used on all slots (tp_xxx functions) yet. So some functions cannot get the defining class.